### ISSUE TITLE
how to train with two outputs?

### ISSUE THREAD

**Issue by fac2003** [tensorflow/tensor2tensor#183]
This is more a question than an issue. I am using the transformer model to train a sequence to sequence model. So far so good, after some tuning, the model trains and learns the mapping I am interested in. 

Part of the target sequence is a yes/no answer and I would like to recover the probability of each symbol according to the model. I realize that I could possibly extract this from the sentence logits, but it seems overkill and would prefer to just stick a softmax on top of the model (or the ClassLabel modality I see in tensor2tensor). 

I saw that the code base supports multiple problems, but that seems to be to train with multiple input features/target combinations. I don't necessarily want to provide/generate input features twice. Should I create a problem that does not provide any additional input features and put the ClassLabel on top, then train with both the sequence and class problems to recover the probability?
I am sure there is a way, but I can't find a clear example in the code base. Thanks.

**Comment by ChunningDu** [comment]
Have you gotten the answer? I am also confused about it. If you know you it, can you share with me? thank you very much!

**Comment by fac2003** [comment]
No, I did not get an answer and have since moved on. Having to do this manually seemed overkill for a quick evaluation of the framework. I could not find a solution in the code base at the time I looked.

**Comment by lukaszkaiser** [comment]
We show how to do this (and more generally get access to internal tensors) in this unit test:
https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/utils/trainer_utils_test.py#L106

**Comment by Dragonet95** [comment]
Resuming this question almost two years later, is there now an easy way to obtain the classification probabilities? In my case i have a text2class problem with 0/1 and i need to get the label probability, but i'm having many troubles to do so, especially after the training and decoding was made so easy
